vnext/Desktop/HttpResource.cpp

```diff
@@ -42,115 +50,176 @@ void HttpResource::SendRequest(
+    {
+    }
+    else if (!bodyData["base64"].empty())
+    {
+    }
+    else if (!bodyData["uri"].empty())
+    {
       assert(false); // Not implemented.
-    } else {
+    }
+    else
+    {
       // Empty request
     }
   }
 
   m_context.restart();
 
   // Send/Receive request.
-  m_resolver.async_resolve(url->host, url->port, [this, &req](boostecr ec, tcp::resolver::results_type results) {
-    if (ec) {
-      if (m_errorHandler)
-        m_errorHandler(ec.message());
-    } else {
-      boost::asio::async_connect(
-          m_socket, results.begin(), results.end(), [this, &req](boostecr ec, const basic_resolver_iterator<tcp> &) {
-            if (ec) {
-              if (m_errorHandler)
-                m_errorHandler(ec.message());
-            } else {
-              async_write(m_socket, move(req), [this](boostecr ec, size_t size) {
-                if (ec) {
-                  if (m_errorHandler)
-                    m_errorHandler(ec.message());
-                } else {
-                  if (m_requestHandler)
-                    m_requestHandler();
-
-                  async_read(m_socket, m_buffer, m_response, [this](boostecr ec, size_t size) {
-                    if (ec) {
-                      if (m_errorHandler)
-                        m_errorHandler(ec.message());
-                    } else {
-                      if (m_responseHandler)
-                        m_responseHandler(boost::beast::buffers_to_string(m_buffer.data()));
-
-                      boost::system::error_code bec;
-                      m_socket.shutdown(tcp::socket::shutdown_both, bec);
-                      if (bec && boost::system::errc::not_connected != bec) // not_connected may happen. Not
-                      // an actual error.
-                      {
-                        if (m_errorHandler)
-                          m_errorHandler(bec.message());
-
-                        // ISS:2306365 - Callback?
-                      }
-                    }
-                  }); // async_read
-                }
-              }); // async_write
-            }
-          }); // async_connect
-    }
-  }); // async_resolve
+  m_resolver.async_resolve(
+    url->host,
+    url->port,
+    bind_front_handler(&HttpResource::OnResolve, this)
+  );
 
   m_context.run();
 }
 
-void HttpResource::AbortRequest() noexcept {
+void HttpResource::OnResolve(error_code ec, tcp::resolver::results_type results)
+{
+  if (ec)
+  {
+    if (m_errorHandler)
+      m_errorHandler(ec.message());
+
+    return;
+  }
+
+  //m_stream.expires_after(std::chrono::seconds(30));//TODO
+
+  m_stream.async_connect(
+    results,
+    bind_front_handler(&HttpResource::OnConnect, this)
+  );
+}
+
+void HttpResource::OnConnect(error_code ec, tcp::resolver::results_type::endpoint_type endpoints)
+{
+  if (ec)
+  {
+    if (m_errorHandler)
+      m_errorHandler(ec.message());
+
+    return;
+  }
+
+  //m_stream.expires_after(std::chrono::seconds(30));//TODO
```

Comment:
>TODO [](start = 55, length = 4)

(nit) are you doing this as part of this cr or separate work item? If separate work item, please add work item id

Reply:
Whether or not I add this, will be determined in this PR (at which point i'll remove the [DO NOT MERGE YET] annotation).

This is a new feature and will research/discuss internally whether it's needed or convenient.